### PR TITLE
Rework java logging and trace correlation

### PR DESCRIPTION
**Pull request by randomanderson**
### What does this PR do?
This updates the documentation for how to configure Java logs to be sent to Datadog and how to correlate logs and traces.

- Remove duplication between the pages and use better hyperlinks
- Consistent use of full file vs partial file for configuration
- Consistent library naming (eg most instances of Slf4j should be logback)
- Fix incorrect config

### Motivation
I worked on a PR for the Java tracer that tests all of the logging library combinations: https://github.com/DataDog/dd-trace-java/pull/2368

When reading the documentation, I noticed it was confusing and outright wrong in some places

### Preview

https://docs-staging.datadoghq.com/landerson/java-log-correlation/logs/log_collection/java/
https://docs-staging.datadoghq.com/landerson/java-log-correlation/tracing/connect_logs_and_traces/java/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/connect_logs_and_traces/java.md [hunk 1]
```diff
@@ -16,43 +16,22 @@ further_reading:
...
-
-Enable injection in the Java tracer's [configuration][1] by adding `-Ddd.logs.injection=true` as a jvm startup argument or through environment variable `DD_LOGS_INJECTION=true`.
-
-If your logs are JSON formated and you are using Logback there is nothing left to do. Otherwise with other logging libraries you need to activate MDC attributes autoinjection into your logs.
-
-If your logs are raw formatted, update your formatter to include `dd.trace_id` and `dd.span_id` in your logger configuration:
-
-{{< tabs >}}
-{{% tab "Log4j2" %}}
+The first step to correlating logs and traces is to [configure your logger][4].  Follow the configuration instructions for your specific logging framework.
```
Comment (andrewsouthard1) [inline content/en/tracing/connect_logs_and_traces/java.md:21]: Can we call out here specifically the loggers we support or that you need to configure - something like:
"Follow the config instructions for log4j, log4j2, or logback"

File: content/en/logs/log_collection/java.md [hunk 2]
```diff
@@ -53,244 +53,208 @@ Here are setup examples for the `log4j`, `slf4j` and `log4j2` logging libraries:
...
-[1]: https://gist.github.com/NBParis/8bda7aea745987dd3261d475c613cf66
-[2]: http://logback.qos.ch/manual/mdc.html
 {{% /tab %}}
-{{% tab "Slf4j" %}}
-
-The JSON library we recommend for Logback is [logstash-logback-encoder][1]. One advantage is: it's inside the main Maven repository.
+{{% tab "Logback" %}}
 
-To add it into your classpath, simply add the following dependency (version 4.5.1 on the example) in your `pom.xml` file:
+We recommend using the [logstash-logback-encoder][1] for JSON formatted logs in Logback. Configure a file appender using the JSON layout in `logback.xml`:
```
Comment (jtappa) [inline content/en/logs/log_collection/java.md:220]:
```suggestion
You can use the [logstash-logback-encoder][1] for JSON formatted logs in Logback. To configure a file appender using the JSON layout in `logback.xml`:
```

File: content/en/logs/log_collection/java.md [hunk 3]
```diff
@@ -53,244 +53,208 @@ Here are setup examples for the `log4j`, `slf4j` and `log4j2` logging libraries:
...
-If APM is enabled for this application and you wish to improve the correlation between application logs and traces, [follow these instructions][2] to set the trace and span ids with [MDC (Mapped Diagnostic Contexts)][3] that are then automatically added in the JSON logs.
-
 [1]: https://github.com/logstash/logstash-logback-encoder
-[2]: /tracing/connect_logs_and_traces/java/
-[3]: http://logback.qos.ch/manual/mdc.html
 {{% /tab %}}
 {{< /tabs >}}
 
+**Inject trace IDs in your logs**
+If APM is enabled for this application, you can correlate logs and traces by [following these instructions][6] to enable trace id injection.
```
Comment (jtappa) [inline content/en/logs/log_collection/java.md:255]:
```suggestion
If APM is enabled for this application, you can correlate logs and traces by referring to [Connecting Java Logs and Traces][6] to enable trace id injection.
```

File: content/en/logs/log_collection/java.md [hunk 4]
```diff
@@ -319,54 +283,60 @@ It is possible to stream logs from your application to Datadog or to the Datadog
...
-* The recommended logging library to stream logs directly is Logback [logstash-logback-encoder][3].
-
-This logging library can be linked from the most common Java ones:
+Most common logging libraries can be bridged to Logback
 
 {{< tabs >}}
 {{% tab "Log4j" %}}
 
-Logging to a remote server in JSON may be difficult with Log4j. It is recommended that you use a SLF4J ship with a module called `log4j-over-slf4j` and then use Logback for the JSON format.
+We advise you to use an slf4j module called [log4j-over-slf4j][1] along with Logback.
```
Comment (jtappa) [inline content/en/logs/log_collection/java.md:295]:
```suggestion
Logging to a remote server in JSON may be difficult. You can use the slf4j module [log4j-over-slf4j][1] with Logback to simplify this process.
```

File: content/en/logs/log_collection/java.md [hunk 5]
```diff
@@ -319,54 +283,60 @@ It is possible to stream logs from your application to Datadog or to the Datadog
...
+[2]: http://www.slf4j.org/legacy.html#log4j-over-slf4j
 {{% /tab %}}
 
 {{% tab "Log4j2" %}}
 
-Log4j2 allows logging to a remote host, but it does not offer the ability to prefix the logs by an API key. Because of this, it is recommended that you use a SLF4J ship with a module called `log4j-over-slf4j` and then use Logback for the JSON format.
+Log4j2 allows logging to a remote host, but it does not offer the ability to prefix the logs by an API key. Because of this, it is recommended that you use a SLF4J  module called `log4j-over-slf4j` and Logback.
 
-To use `log4j-over-slf4j` in your own application, the first step is to find-and-replace `log4j.jar` with `log4j-to-slf4j-2.11.jar`.
+`log4j-to-slf4j.jar` seamlessly replaces log4j2 in your application so you do not have to make any code changes.  To use it
```
Comment (jtappa) [inline content/en/logs/log_collection/java.md:333]:
```suggestion
`log4j-to-slf4j.jar` seamlessly replaces log4j2 in your application so you do not have to make any code changes. To use it:
```

File: content/en/tracing/connect_logs_and_traces/java.md [hunk 6]
```diff
@@ -16,43 +16,22 @@ further_reading:
       tag: 'Blog'
       text: 'Correlate request logs with traces automatically'
 ---
+## Configure Your Logger
```
Comment (jtappa) [inline content/en/tracing/connect_logs_and_traces/java.md:19]:
```suggestion
## Configure your logger
```

File: content/en/tracing/connect_logs_and_traces/java.md [hunk 7]
```diff
@@ -16,43 +16,22 @@ further_reading:
...
-{{% tab "Log4j2" %}}
+The first step to correlating logs and traces is to configure your logger.  Follow the [configuration instructions][4] for Log4j, Log4j2, or Logback.
 
-```xml
-<Pattern>"%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %X{dd.trace_id} %X{dd.span_id} - %m%n"</Pattern>
-```
-
-{{% /tab %}}
-{{% tab "slf4j/logback" %}}
+## Automatically Inject Trace and Span IDs
```
Comment (jtappa) [inline content/en/tracing/connect_logs_and_traces/java.md:23]:
```suggestion
## Automatically inject trace and span IDs
```

File: content/en/logs/log_collection/java.md [hunk 8]
```diff
@@ -319,54 +283,60 @@ It is possible to stream logs from your application to Datadog or to the Datadog
...
 1. Add the Logback logging library to your code (or build a bridge from your current logger to it)
-2. Configure it to send logs to Datadog
+2. Configure Logback to send logs to Datadog
 
 ### Bridge from Java logging libraries to Logback
 
-* The recommended logging library to stream logs directly is Logback [logstash-logback-encoder][3].
-
-This logging library can be linked from the most common Java ones:
+Most common logging libraries can be bridged to Logback
```
Comment (jtappa) [inline content/en/logs/log_collection/java.md:290]: What does this mean, can you include more context? Also looks like this needs a period to be a complete statement.

**Review by jtappa — CHANGES_REQUESTED** [review]

File: content/en/logs/log_collection/java.md [hunk 9]
```diff
@@ -436,40 +390,61 @@ Configure the Logback logger to stream logs directly to Datadog by adding the fo
...
+  </appender>
+  <appender name="JSON_TCP" class="net.logstash.logback.appender.LogstashTcpSocketAppender">
     <remoteHost>tcp-intake.logs.datadoghq.eu</remoteHost>
     <port>1883</port>
     <keepAliveDuration>20 seconds</keepAliveDuration>
     <encoder class="net.logstash.logback.encoder.LogstashEncoder">
         <prefix class="ch.qos.logback.core.encoder.LayoutWrappingEncoder">
             <layout class="ch.qos.logback.classic.PatternLayout">
-                <pattern><API_KEY> %mdc{keyThatDoesNotExist}</pattern>
+                <pattern><APIKEY> %mdc{keyThatDoesNotExist}</pattern>
```
Comment (randomanderson) [inline content/en/logs/log_collection/java.md:418]:
```suggestion
                <pattern><API_KEY> %mdc{keyThatDoesNotExist}</pattern>
```

File: content/en/logs/log_collection/java.md (hunk 8 above)
Comment (randomanderson) [reply content/en/logs/log_collection/java.md:290]: This is connected to the "There are two steps to configure the Java application to..." section right before.

How about:

```suggestion
If you are not using Logback as your logging library, the first step to stream logs directly to Datadog is to bridge your logging library to Logback. The exact steps depends on your logging library:
```

**Review by kayayarai — APPROVED** [review]

**Review by jtappa — APPROVED** [review]

**Pull request merged**